test/test_browser.py

```diff
@@ -5549,6 +5554,58 @@ def test_webpack(self):
     shutil.copyfile('webpack/src/hello.wasm', 'webpack/dist/hello.wasm')
     self.run_browser('webpack/dist/index.html', '/report_result?exit:0')
 
+  def test_fetch_polyfill_shared_lib(self):
+    create_file('library.c', r'''
+      #include <stdio.h>
+      int library_func() {
+        return 42;
+      }
+    ''')
+    create_file('main.c', r'''
+      #include <assert.h>
+      #include <dlfcn.h>
+      #include <stdio.h>
+      #include <emscripten.h>
+      int main() {
+        int found = EM_ASM_INT(
+          return preloadedWasm['/library.so'] !== undefined;
+        );
+        void *lib_handle = dlopen("/library.so", RTLD_NOW);
+        typedef int (*voidfunc)();
+        voidfunc x = (voidfunc)dlsym(lib_handle, "library_func");
+        printf("%d\n", x());
+        return 0;
+      }
+    ''')
+    create_file('report_result.html', r'''
+      <html>
+      <body>
+        <script type='text/javascript'>
+          var Module = {
+            print: (text) => fetch('http://localhost:8888/report_result?'+ (text.includes("42") ? 1 : 0)
+            )
+          };
+        </script>
+        {{{ SCRIPT }}}
+      </body>
+      </html>
+    ''')
+
+    self.run_process([EMCC, 'library.c', '-sSIDE_MODULE', '-O2', '-o', 'library.so'])
+
+    def test(args, expect_fail):
+      self.compile_btest('main.c', ['library.so', '-sMAIN_MODULE', '--shell-file', 'report_result.html', '-o', 'a.out.html'])
```

Comment:
Can you use `-sMAIN_MODULE=2` here?   This makes the test program a lot smaller.